schema/Metrics.v1.0.0.schema.json

```
@@ -0,0 +1,354 @@
+          "exclusiveMaximum": true,
+          "description": "The percentile to measure."
+        }
+      },
+      "description": "The definition of a Percentile metric. This metric kind measures the percentile of an event property."
+    },
+    "SumDefinition": {
+      "type": "object",
+      "required": ["kind", "value"],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "enum": ["Sum"],
+          "description": "The kind of metric."
+        },
+        "value": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/AggregatedValue"
+            }
+          ],
+          "description": "The value to aggregate."
+        }
+      },
+      "description": "The definition of a Sum metric. This metric kind measures the sum of an event property. Experiment analysis accounts for unequal traffic allocation."
+    },
+    "AverageDefinition": {
+      "type": "object",
+      "required": ["kind", "value"],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "enum": ["Average"],
+          "description": "The kind of metric."
+        },
+        "value": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/AggregatedValue"
+            }
+          ],
+          "description": "The value to aggregate."
+        }
+      },
+      "description": "The definition of an Average metric. This metric kind measures the average of an event property."
+    },
+    "ObservedEvent": {
+      "type": "object",
+      "description": "An event observed by a metric.",
+      "properties": {
+        "eventName": {
+          "type": "string",
+          "description": "The name of the event.",
+          "minLength": 1,
+          "maxLength": 200
+        },
+        "filter": {
+          "$ref": "#/definitions/EventCondition",
+          "description": "[Optional] A condition to filter events."
+        }
+      },
+      "required": ["eventName"]
+    },
+    "EventCondition": {
+      "type": "string",
+      "description": "A condition to filter events. Accepts a Kusto Query Language (KQL) filter predicate. \nAllowed expressions evaluate to a bool data type and use a subset of KQL syntax:\nEvent properties: plain or bracket notation.\nLiterals: bool, long, real, string.\nComparison operators: ==, !=, <, <=, >, >=.\nBoolean operators: and, or.\nGrouping operations: parentheses.",
+      "minLength": 1,
+      "maxLength": 1000
+    },
+    "AggregatedValue": {
+      "type": "object",
+      "required": ["eventName", "eventProperty"],
+      "properties": {
+        "eventName": {
+          "type": "string",
+          "minLength": 1,
+          "maxLength": 200,
+          "description": "The name of the event."
+        },
+        "filter": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/EventCondition"
+            }
+          ],
+          "description": "[Optional] A condition to filter events."
+        },
```

Why use `allOf` here?
```suggestion
        "filter": {
          "$ref": "#/definitions/EventCondition",
          "description": "[Optional] A condition to filter events."
        },
```

---

Removed